src/main/java/TestsService.java

```diff
@@ -22,8 +24,9 @@ public TestsService(TaskDao taskDao) {
     public PerformedTask submitTask(long studentId, long taskId, double calculatedResult) {
         logger.info("Submitting task for studentId {} taskId {} and calculated result {}", studentId, taskId, calculatedResult);
         Task task = taskDao.getById(taskId);
-        boolean isPassed = calculateIfResultIsCorrect(task.getExpectedResult(), calculatedResult, task.getMarginOfError());
+        boolean isPassed = calculateIfResultIsCorrect(task.getExpectedResult(), calculatedResult, task.getMarginOfError()); // Pytanie! Dlaczego nie przekazujemy poprostu całego obiektu klasy task, tylko wartości double jako wyniki metod klasy task? Wtedy przekazywalibyśmy dwa argumenty zamiast trzech
         //TODO(instead of just returning object here, we have to save it to the database)
+        taskDao.save(task); // czy zamieniać metodę na typu void ? jeśli tak zrobimy to testy trzeba będzie zmodyfikować
```

Comment:
ja bym zostawił tak jak jest tylko wtedy trzeba zmienić fakeTaskDao dla pierwszego testu i TestFakeImpl (dla testowania poprzez @Spy)
Ogólnie to normalne, że jak dodajesz jakąś funkcjonalność to testy się wywalają. Ale to jest też zaleta testów bo zmusza Cię do zastanowienia się dlaczego się wywaliły i czy powinny były się wywalić.